### ISSUE TITLE
Setting `currentUser` on email/password signup with automatic login after signup

### ISSUE THREAD

**Issue by bartriordan** [sahat/instagram-hackhands#4]
tl;dr: Email/password signup with automatic login doesn't save a `currentUser` value and causes an error on page refresh. Can be fixed with slight change to `SignupCtrl`.

I used this repository and the associated tutorial to implement an email/password signup system, but noticed a problem that can be recreated as follows: 
1. A user signs up with an email and password, and then are automatically logged in and redirected to the home page;
2. Satellizer saves the JWT to `$window.localStorage` on the call to its `local.signup` function (via the call to `$auth.signup`), but does not create a `currentUser` value in either `localStorage` or the `$rootScope`;
3. The signed-up and automatically-logged-in user refreshes the page or closes the page and opens it again later;
4. On reload, the `run` block in `app.js` confirms that there is an unexpired JWT saved to `localStorage` and attempts to access the property `$window.localStorage.currentUser`, which has never been set and is therefore `undefined`;
5. The `run` block calls `JSON.parse` on the `undefined` property, which causes an `Uncaught SyntaxError: Unexpected token u`.

This issue can be fixed by slightly changing `SignupCtrl` to more-closely match the pattern used in `LoginCtrl`. The current signup function is:

```
$scope.signup = function() {
  var user = {
    email: $scope.email,
    password: $scope.password
  };

  $auth.signup(user)
    .catch(function(response) {
      console.log(response.data);
    });
};
```

Adding a `then` block after the call to `$auth.signup` solves the problem:

```
$scope.signup = function() {
  var user = {
    email: $scope.email,
    password: $scope.password
  };

  $auth.signup(user)
    .then(function(response) {
      $window.localStorage.currentUser = JSON.stringify(response.data.user);
      $rootScope.currentUser = JSON.parse($window.localStorage.currentUser);
    })

    .catch(function(response) {
      console.log(response.data);
    });
};
```

This way, as soon as a user is successfully signed up the value of `currentUser` is stored in both `$window.localStorage` and the `$rootScope`.
